extension/src/popup/components/manageAssets/AddToken/index.tsx

```diff
@@ -145,17 +146,26 @@ export const AddToken = () => {
 
           const res = await fetch(tokenUrl.href);
           const resJson = await res.json();
-
-          setAssetRows([
-            {
-              code: resJson.symbol,
-              issuer: contractId,
-              domain: "",
-              name: resJson.name,
-            },
-          ]);
+          if (!res.ok) {
+            captureException(
+              `Failed to fetch token details - ${JSON.stringify(resJson)}`,
+            );
+            setAssetRows([]);
+          } else {
+            setAssetRows([
+              {
+                code: resJson.symbol,
+                issuer: contractId,
+                domain: "",
+                name: resJson.name,
+              },
+            ]);
+          }
         } catch (e) {
           setAssetRows([]);
+          captureException(
+            `Failed to fetch token details - ${JSON.stringify(e)}`,
```

Comment:
Maybe we should surface this error to users - let them know their assets aren't gone, we just couldn't fetch them

Reply:
similar to what you're doing below with Account Balances

Reply:
hmm yeah so I think we could show something like "Token unknown" instead of the token details?

Reply:
so thinking about this more, I think in the case that we fail to fetch token details we can do a couple of things -
We can surface the error and let them add it still, and label it "unknown" or something.
We can surface the error and not let them add the token.

We don't really have a pattern for inline errors so I'm not super sure what we would want here display wise. I would imagine an inline error and either one of the options above but let me know if you had something else in mind.